server/handlers/logic.go

```diff
@@ -0,0 +1,89 @@
+package handlers
+
+import (
+	"io"
+
+	"github.com/Sirupsen/logrus"
+	ctxu "github.com/docker/distribution/context"
+	"golang.org/x/net/context"
+
+	"github.com/docker/notary/server/errors"
+	"github.com/docker/notary/server/snapshot"
+	"github.com/docker/notary/server/storage"
+	"github.com/docker/notary/server/timestamp"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+func getRole(ctx context.Context, logger ctxu.Logger, w io.Writer, store storage.MetaStore, gun, role, checksum string) error {
+	var (
+		out []byte
+		err error
+	)
+	if checksum == "" {
+		// the timestamp and snapshot might be server signed so are
+		// handled specially
+		switch role {
+		case data.CanonicalTimestampRole, data.CanonicalSnapshotRole:
+			return getMaybeServerSigned(ctx, w, logger, store, gun, role)
+		}
+		out, err = store.GetCurrent(gun, role)
+	} else {
+		out, err = store.GetChecksum(gun, role, checksum)
+	}
+
+	if err != nil {
+		if _, ok := err.(storage.ErrNotFound); ok {
+			logrus.Errorf("404 GET %s:%s@%s, error: %v", gun, role, checksum, err)
+			return errors.ErrMetadataNotFound.WithDetail(nil)
+		}
+		logger.Error("500 GET")
+		return errors.ErrUnknown.WithDetail(err)
+	}
+	if out == nil {
+		logger.Error("404 GET")
```

Comment:
If the logger is already going to log the information, does having an additional `logger.Error` here helpful for debugging?  (I guess on the terminal it's red, so easier to see).